app/models/imo_delivery_detail.rb

```diff
@@ -0,0 +1,32 @@
+class ImoDeliveryDetail < ApplicationRecord
+  has_one :communication, as: :detailable
+
+  enum result: {
+    error: "error",
+    no_imo_account: "no_imo_account",
+    not_subscribed: "not_subscribed",
+    read: "read",
+    sent: "sent"
+  }
+
+  after_create :update_authorization, if: proc { |detail| detail.result.in? ["not_subscribed", "no_imo_account"] }
+
+  def unsuccessful?
+    error? || no_imo_account? || not_subscribed?
+  end
+
+  def successful?
+    read?
+  end
+
+  def in_progress?
+    sent?
+  end
+
+  private
+
+  def update_authorization
```

Comment:
Update the patient's imo_authorization if the notification failed due to the patient unsubscribing or deleting their account.